src/tests/ftest/util/command_utils_base.py

```diff
@@ -670,22 +679,25 @@ def get_export_str(self, separator=";"):
         return export_str
 
 
-class PositionalParameter(BasicParameter):
+class PositionalParameter(FormattedParameter):
     """Parameter that defines position.
 
     Used to support positional parameters for dmg and daos.
     """
 
-    def __init__(self, position, default=None):
-        """Create a PositionalParameter  object.
+    def __init__(self, position, default=None, str_format=None, yaml_key=None):
+        """Create a PositionalParameter object.
 
         Args:
             position (int): argument position/order
             default (object, optional): default value for the param. Defaults to
                 None.
 
         """
-        super().__init__(default, default)
+        if str_format is None:
+            str_format = "{}"
```

Comment:
This is a little odd, why this rather than making "{}" the default?

Reply:
This is because the `str_format` is optional.  It is used to format the PositionalParameter's string representation in https://github.com/daos-stack/daos/blob/763eb9b740a1d41c70b07f32dcec013319c30d1f/src/tests/ftest/util/command_utils_base.py#L159-L168 

This allows us to represent different styles of command arguments with the one class, for example:

- "-v" would be represented by `PositionalParameter(1, True, "-v")`, where setting PositionalParameter.value = true / false toggles whether or not to include the "-v" command argument
- "--file=/path/to/file" would be represented by `PositionalParameter(2, "/path/to/file", "--file={}")`
- "78" would be represented by `PositionalParameter(1, 78, None)`

Reply:
That's a reasonable suggestion

Reply:
I've been trying to think of a reason why it's logically different and the only case I can think of is if something is passing in str_format=None for some reason which seems unlikely but is possible.

Reply:
I made `"{}"` as default.